### PR TITLE
Ret 337

### PR DESCRIPTION
**Pull request by pmumbere**
### JIRA link ###

[RET-337](https://tools.hmcts.net/jira/browse/RET-337)

### Change description ###

This PR implements the return to claim page 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: nodemon.json [hunk 1]
```diff
@@ -1,6 +1,10 @@
 {
-  "watch": ["src/main"],
-  "ext": "ts,js",
+  "ext": "ts,js,json",
```
Comment (tautvydas-cerniauskas) [inline nodemon.json:2]: Is there a reason why you made changes to this file? Seems kinda unnecessary.

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "start": "NODE_ENV=production ./node_modules/.bin/ts-node -r tsconfig-paths/register src/main/server.ts",
     "start:dev": "nodemon --config nodemon.json",
+    "start:local": "nodemon --ext ts,js,json --exec ts-node ./src/main/server.ts",
```
Comment (tautvydas-cerniauskas) [inline package.json:11]: I'm almost certain that `start:dev` does the exactly same thing.

**Review by jthmcts — COMMENTED** [review]
There's a lot of seemingly unnecessary whitespace in several files. Can you tidy this up?

File: src/test/unit/mocks/mockRequest.ts [hunk 3]
```diff
@@ -26,3 +26,29 @@ export const mockRequest = ({
   } as any;
   return req;
 };
+
+export const returnToExistingMockRequest = ({
```
Comment (tautvydas-cerniauskas) [inline src/test/unit/mocks/mockRequest.ts:30]: You don't need to create a new one, the original object can be extended with new values.

File: src/main/controllers/return_to_existing_claim/ReturnToExistingController.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { Response } from 'express';
+import { Form } from '../../components/form/form';
+import { FormContent, FormFields } from '../../definitions/form';
+import { AppRequest } from '../../definitions/appRequest';
```
Comment (tautvydas-cerniauskas) [inline src/main/controllers/return_to_existing_claim/ReturnToExistingController.ts:4]: Could you please tidy up white spaces and new lines across all the files?

File: src/test/routes/return-to-existing.test.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { expect } from 'chai';
+import request from 'supertest';
+
+import { app } from '../../main/app';
+
+describe('GET /return-to-existing', () => {
+  it('should return the return to existing page', async () => {
```
Comment (marcjudge) [inline src/test/routes/return-to-existing.test.ts:7]: 'return the return' looks like a typo. Please clarify the should statement

File: src/test/routes/return-to-existing.test.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    const res = await request(app).get('/return-to-existing');
+    expect(res.type).to.equal('text/html');
+    expect(res.status).to.equal(200);
+  });
+});
+
+describe('on POST /return-to-existing', () => {
+  test('should redirect to home page when an option is selected is selected', async () => {
```
Comment (marcjudge) [inline src/test/routes/return-to-existing.test.ts:15]: 'is selected' twice

File: package.json (hunk 2 above)
Comment (pmumbere) [reply package.json]: reverted file

**Review by tautvydas-cerniauskas — CHANGES_REQUESTED** [review]

File: src/test/unit/mocks/mockRequest.ts [hunk 7]
```diff
@@ -17,6 +17,7 @@ export const mockRequest = ({
     userCase: {
       id: '1234',
       dobDate: { year: '2000', month: '12', day: '24' },
+      return_number_or_account: { have_return_number: '', have_account: '' },
```
[the remark this answers is not in the document]
Comment (pmumbere) [reply src/test/unit/mocks/mockRequest.ts]: Done

File: src/test/routes/return-to-existing.test.ts (hunk 6 above)
Comment (pmumbere) [reply src/test/routes/return-to-existing.test.ts:14]: amended

File: src/test/routes/return-to-existing.test.ts (hunk 5 above)
Comment (pmumbere) [reply src/test/routes/return-to-existing.test.ts:6]: amended

**Review by marcjudge — APPROVED** [review]

File: src/test/unit/mocks/mockRequest.ts [hunk 8]
```diff
@@ -16,7 +16,7 @@ export const mockRequest = ({
   req.session = {
     userCase: {
       id: '1234',
-      dobDate: { year: '2000', month: '12', day: '24' },
+      dobDate: { year: '2000', month: '12', day: '24' },
```
Comment (tautvydas-cerniauskas) [inline src/test/unit/mocks/mockRequest.ts:19]: These changes seem irrelevant.

**Review by tautvydas-cerniauskas — COMMENTED** [review]
Similar issues to RET-845. Please refer to those comments.

**Review by tautvydas-cerniauskas — APPROVED** [review]
